src/cmipcite/citations.py

```diff
@@ -70,15 +70,29 @@ def get_citation_for_tracking_id(
     Returns
     -------
     :
-        Citation for the given `tracking_id`
+        Citation for the given `tracking_id` or PID
     """
     client = RESTHandleClient(handle_server_url="http://hdl.handle.net/")
 
-    tracking_id_query = tracking_id.replace("hdl:", "")
-
-    # handle_ds = client.get_value_from_handle(tracking_id_query, "IS_PART_OF")
-    version = client.get_value_from_handle(tracking_id_query, "VERSION_NUMBER")
-    doi = client.get_value_from_handle(tracking_id_query, "IS_PART_OF")
+    id_query = input_id.replace("hdl:", "")
+
+    agg_lev = client.get_value_from_handle(id_query, "AGGREGATION_LEVEL")
+
+    # if the input is a pid (associated to a dataset), the is_part_of is a doi.
+    if agg_lev == "DATASET":
+        doi = client.get_value_from_handle(id_query, "IS_PART_OF")
+        version = client.get_value_from_handle(id_query, "VERSION_NUMBER")
+    # if the input is a tracking_id (associated to a file),
+    # the is_part_of is a pid of the dataset.
+    # and we need an extra step to get the doi.
+    elif agg_lev == "FILE":
+        pid = client.get_value_from_handle(id_query, "IS_PART_OF")
+        doi = client.get_value_from_handle(pid, "IS_PART_OF")
+        version = client.get_value_from_handle(pid, "VERSION_NUMBER")
+    else:
+        raise NotImplementedError(
+            f"The id passed has an unknown AGGREGATION_LEVEL: {agg_lev}"
+        )
```

Comment:
```suggestion
    else:
        raise NotImplementedError(
            f"The id {input_id} has an unknown AGGREGATION_LEVEL: {agg_lev}"
        )
```

I would add the input ID too - just so that if you hit this error in a loop or whatever, you can see exactly which ID failed